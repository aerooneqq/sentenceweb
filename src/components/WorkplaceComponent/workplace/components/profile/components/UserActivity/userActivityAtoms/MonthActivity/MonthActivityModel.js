import React from "react";

import SingleUserActivity from "../SingleUserActivity/SingleUserActivity";

export default class MonthActivityModel { 
    days = ["Mn", "Tu", "Wd", "Th", "Fr", "St", "Sn"];

    constructor(activities, daysCount, firstDayNum) { 
        this._tdKey = 0;
        this._trKey = 0;

        this._activities = activities;
        this._daysCount = daysCount;
        this._firstDayNum = firstDayNum;
    } 

    getActivitiesTable(singleActivityClickFunc) { 
        let trList = [];
        let tdList = [];

        for (let i = 0; i < this.days.length; i++) { 
            tdList.push(<td key = {this._getTdKey()}> {this.days[i]} </td>)
        }

        trList.push(<tr key = {this._getTrKey()} className="dayName">{tdList}</tr>)

        tdList = []

        for (let i = 0; i < this._daysCount + this._firstDayNum; i++) {
            if (i % 7 === 0) {
                trList.push(<tr key = {this._getTrKey()}>{tdList}</tr>);
                tdList = []
            } 

            if (i < this._firstDayNum) { 
                tdList.push(<td key = {this._getTdKey()}></td>)
            }
            else { 
                tdList.push(
                    <td key = {this._getTdKey()}>
                        <SingleUserActivity activityData = {this._activities[i - this._firstDayNum]} 
                                            handleSingleUserActivityClick = {singleActivityClickFunc}/>
                    </td>)
            }
        }

        trList.push(<tr key = {this._getTrKey()}>{tdList}</tr>)

        return trList;
    }

    _getTrKey() { 
        return this._trKey++;
    }

    _getTdKey() { 
        return this._tdKey++;
    }
}