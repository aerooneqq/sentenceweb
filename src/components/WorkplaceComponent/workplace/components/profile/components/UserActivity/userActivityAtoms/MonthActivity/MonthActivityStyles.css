.monthNameCont{ 
    font-family: Arial;
    font-size: 15px;
    color: black;
    margin-left: 5px;

    cursor: default;

    z-index: 1;
}

.calendarContainer { 
    width: 100%;
    height: 100%;

    position: relative;
    top: 0;
    left: 0;

    z-index: 1;
}

.userActivityDataContainer { 
    width: 100%;
    height: 100%;
    background: #f9f9f9;

    position: relative;
    top: 0;
    left: 0;

    z-index: 2; 
}
    .userActivityDataContFadeIn { 
        animation: slideUpAnimation 0.2s linear forwards,
                   fadeInAnimation 0.3s linear forwards;
    }

    @keyframes slideUpAnimation { 
        from { 
            top: 0
        }

        to { 
            top: -101%;
        }
    }

    @keyframes fadeInAnimation { 
        from { 
            opacity: 0;
        }

        to { 
            opacity: 1;
        }
    }

.monthActivityCont{ 
    margin-top: 10px;
}

.monthActivityCont:nth-child(1){ 
    margin-left: 5px;
}

.monthActivityCont:nth-last-child(1){ 
    margin-right: 5px;
}

.dayName{ 
    font-family: Arial; 
    color: black;
    font-size: 12px;

    cursor: default;
}