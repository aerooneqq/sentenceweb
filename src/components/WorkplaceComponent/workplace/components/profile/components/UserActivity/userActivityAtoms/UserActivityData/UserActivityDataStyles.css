.userActivityData { 
    width: 100%;
    height: 100%;

    background: transparent;
}

.userActivityDataHeaderCont { 
    width: 100%;
    height: auto;

    box-sizing: border-box;

    padding-left: 5%;
    padding-right: 5%;

    display: flex;
    flex-direction: row;

    justify-content: center;
    align-items: center;
}

.userActivityDataHeaderText { 
    font-size: 14px;
    color: black;
}

.userActivityDataFill { 
    flex-grow: 1;
}

.closeUserActivityIcon { 
    width: 20px;
    height: 20px;

    margin-top: 5px;
}