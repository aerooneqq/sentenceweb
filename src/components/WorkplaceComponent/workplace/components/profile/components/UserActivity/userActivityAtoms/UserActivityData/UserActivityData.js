import React, {Component} from "react"

//Styles
import "./UserActivityDataStyles.css"

//Icons
import closeIconStatic from "./img/close_user_activity_data_icon.png";
import closeIconActive from "./img/close_user_activity_data_icon_active.png"
import UserActivitiesSingleData from "./UserActivitySingleData/UserActivitySingleData";

/**
 * PROPS LIST:
 * 1) activityData
 * 2) closeActivityData - hides the activity data
 */
export default class UserActivityData extends Component { 

    constructor(props) { 
        super(props);

        this.state = { 
            isCloseIconActive: false
        };

        this.onCloseIconMouseEnter = this.onCloseIconMouseEnter.bind(this);
        this.onCloseIconMouseLeave = this.onCloseIconMouseLeave.bind(this);
    }

    onCloseIconMouseEnter() { 
        this.setState({ 
            isCloseIconActive: true
        });
    }

    onCloseIconMouseLeave() { 
        this.setState({ 
            isCloseIconActive: false
        });
    }
    
    render() { 
        return ( 
            <div className = "userActivityData">
                <div className = "userActivityDataHeaderCont">
                    <div className = "userActivityDataHeaderText">
                        Activity data
                    </div>
                    <div className = "userActivityDataFill" />
                    <div className = "userActivityDataHeaderIcon">
                        <img src = {this.state.isCloseIconActive === true ? closeIconActive : closeIconStatic } 
                             alt = "Close"
                             className = "closeUserActivityIcon"
                             onMouseEnter = {this.onCloseIconMouseEnter}
                             onMouseLeave = {this.onCloseIconMouseLeave}
                             onClick = {this.props.closeActivityData} />
                    </div>
                </div>
                
                <div className = "singleUserActivitiesCont">
                    {this.props.activityData.activities.map(activityInfo => <UserActivitiesSingleData activity = {activityInfo}/>)}
                </div>
            </div>
        )
    }
}