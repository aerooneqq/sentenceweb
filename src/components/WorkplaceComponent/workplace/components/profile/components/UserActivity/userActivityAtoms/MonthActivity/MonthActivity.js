import React, {Component, lazy} from "react"

import "./MonthActivityStyles.css"

//Components
import MonthActivityModel from "./MonthActivityModel";
import UserActivityData from "../UserActivityData/UserActivityData";

/**
 * PROPS LIST:
 * 1) activitiesData - the data which describes the activity
 * 2) daysCount - the number of days in a month
 * 3) month - the object which is a month description
 */
export default class MonthActivity extends Component{ 

    constructor(props) { 
        super(props);

        this.showUserActivityData = this.showUserActivityData.bind(this);
        this.closeActivityData = this.closeActivityData.bind(this);

        this.state = {
            trList: new MonthActivityModel(props.activitiesData, 
                                           props.daysCount,
                                           props.month.firstDayNum).getActivitiesTable(this.showUserActivityData),
            isExtraDataVisible: false,
            extraData: null
        }
    }

    showUserActivityData(data) { 
        this.setState({ 
            isExtraDataVisible: true,
            extraData: data
        });
    }

    closeActivityData() { 
        this.setState({ 
            isExtraDataVisible: false,
            extraData: null
        });
    }

    render() {
        return(
            <div className = "monthActivityCont">
                <div className = "calendarContainer">
                    <div className = "monthNameCont">{this.props.month.name}</div>
                    <table>
                        {this.state.trList}
                    </table>
                </div>
                {this.state.isExtraDataVisible === true ? 
                <div className = "userActivityDataContainer userActivityDataContFadeIn">
                    <UserActivityData closeActivityData = {this.closeActivityData}
                                      activityData = {this.state.extraData} />
                </div> : null}
            </div>
        )
    }
}