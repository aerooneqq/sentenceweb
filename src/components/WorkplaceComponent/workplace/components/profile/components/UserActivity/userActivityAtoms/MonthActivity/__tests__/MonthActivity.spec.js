import React from "react";
import {shallow} from "enzyme";
import {shallowToJson} from "enzyme-to-json";
import MonthActivity from "../MonthActivity";
import UserActivityData from "../../UserActivityData/UserActivityData";
import { monitorEventLoopDelay } from "perf_hooks";

const props = { 
    activitiesData: 3,
    daysCount: 30, 
    month: { 
        firstDayNum: 4,
        name: "August"
    }
}

describe("MonthActivity", () => { 
    it("should render correctly", () => { 
        expect(shallowToJson(shallow(<MonthActivity {...props}/>))).toMatchSnapshot();
    });

    it("should have proper state", () => { 
        const properState = { 
            isExtraDataVisible: false,
            extraData: null
        }

        const monthActivity = shallow(<MonthActivity {...props}/>);

        for (let key in properState) { 
            expect(properState[key]).toBe(monthActivity.state(key));
        }
    });

    it("should diplay additional data when isExtraDataVisible is true", () => { 
        const monthActivity = shallow(<MonthActivity {...props} />);

        monthActivity.setState({ isExtraDataVisible: true });
        expect(monthActivity.find(UserActivityData).exists()).toBe(true);
    });

    it("should display month name from props", () => { 
        const monthActivity = shallow(<MonthActivity {...props}/>);

        expect(monthActivity.find("div.monthNameCont").text()).toBe(props.month.name);
    });
});