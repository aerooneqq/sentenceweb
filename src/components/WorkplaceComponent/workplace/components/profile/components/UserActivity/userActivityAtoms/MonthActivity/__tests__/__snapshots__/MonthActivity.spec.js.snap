// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthActivity should render correctly 1`] = `
<div
  className="monthActivityCont"
>
  <div
    className="calendarContainer"
  >
    <div
      className="monthNameCont"
    >
      August
    </div>
    <table>
      <tr
        className="dayName"
        key="0"
      >
        <td
          key="0"
        >
           
          Mn
           
        </td>
        <td
          key="1"
        >
           
          Tu
           
        </td>
        <td
          key="2"
        >
           
          Wd
           
        </td>
        <td
          key="3"
        >
           
          Th
           
        </td>
        <td
          key="4"
        >
           
          Fr
           
        </td>
        <td
          key="5"
        >
           
          St
           
        </td>
        <td
          key="6"
        >
           
          Sn
           
        </td>
      </tr>
      <tr
        key="1"
      />
      <tr
        key="2"
      >
        <td
          key="7"
        />
        <td
          key="8"
        />
        <td
          key="9"
        />
        <td
          key="10"
        />
        <td
          key="11"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
        <td
          key="12"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
        <td
          key="13"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
      </tr>
      <tr
        key="3"
      >
        <td
          key="14"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
        <td
          key="15"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
        <td
          key="16"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
        <td
          key="17"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
        <td
          key="18"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
        <td
          key="19"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
        <td
          key="20"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
      </tr>
      <tr
        key="4"
      >
        <td
          key="21"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
        <td
          key="22"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
        <td
          key="23"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
        <td
          key="24"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
        <td
          key="25"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
        <td
          key="26"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
        <td
          key="27"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
      </tr>
      <tr
        key="5"
      >
        <td
          key="28"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
        <td
          key="29"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
        <td
          key="30"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
        <td
          key="31"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
        <td
          key="32"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
        <td
          key="33"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
        <td
          key="34"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
      </tr>
      <tr
        key="6"
      >
        <td
          key="35"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
        <td
          key="36"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
        <td
          key="37"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
        <td
          key="38"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
        <td
          key="39"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
        <td
          key="40"
        >
          <SingleUserActivity
            handleSingleUserActivityClick={[Function]}
          />
        </td>
      </tr>
    </table>
  </div>
</div>
`;
